weave/scorers/hallucination_scorer.py

```diff
@@ -157,3 +173,124 @@ async def score(self, output: str, context: str) -> HallucinationResponse:
         )
         return response.model_dump()
+
+
+HALLUCINATION_SCORER_THRESHOLD = 0.35
+
+
+class WeaveHallucinationScorerV1(HuggingFacePipelineScorer):
+    """
+    A scorer that detects hallucinations in the output, given an query and context. This scorer
+    uses the HHEM 2.1 model from Vectara, https://huggingface.co/vectara/hallucination_evaluation_model
+
+    This scorer uses a fine-tuned LLM to analyze whether model outputs contain information not supported
+    by the given context.
+
+    Args:
+        device: Device to run model on, defaults to "cuda"
+        model_name_or_path: Path or name of model weights to load
+
+    Note: This Scorer's `score` method will score the text passed to its `output` parameter for hallucinations
+    based on the query and context.
+
+    Returns:
+        dict: A dictionary containing:
+            - pass (bool): True if no hallucinations detected (score <= threshold)
+            - extras (dict): Contains:
+                - score (float): Hallucination score between 0 and 1, where lower is better
+                - error (str, optional): Error message if something went wrong
+
+    Example:
+        >>> scorer = HallucinationScorer()
+        >>> result = scorer.score(
+        ...     query="What is the capital of France?",
+        ...     context="Paris is the capital of France.",
+        ...     output="Paris is the capital of France."
+        ... )
+        >>> print(result)
+        {
+            'pass': True,
+            'extras': {
+                'score': 0.1
+            }
+        }
+    """
+
+    threshold: float = HALLUCINATION_SCORER_THRESHOLD
+    _model_max_length: int = PrivateAttr(default=8192)
+    task: str = "pair-classification"
+
+    def load_pipeline(self) -> "Pipeline":
+        ensure_hf_imports()
+        from transformers import pipeline
+
+        self.device = set_device(self.device)
+        self._local_model_path = load_hf_model_weights(
+            self.model_name_or_path, MODEL_PATHS["hallucination_scorer"]
+        )
+        # HuggingFacePipelineScorer's model_post_init will cache the pipeline at self._pipeline
+        return pipeline(
+            task=self.task,
+            model=self._local_model_path,
+            device=self.device,
+            trust_remote_code=True,
+        )
+
+    def _predict(
+        self, query: str, context: Union[str, list[str]], output: str
+    ) -> float:
+        tokenizer = self._pipeline.tokenizer
+        context_str = "\n\n".join(context) if isinstance(context, list) else context
+        inps = query + "\n\n" + context_str
+        outs = output
+
+        inps_toks = tokenizer(inps, truncation=False)
+        outs_toks = tokenizer(outs, truncation=False)
+
+        len_inps = len(inps_toks.input_ids)
+        len_outs = len(outs_toks.input_ids)
+
+        # Handle large inputs
+        if len_inps + len_outs > self._model_max_length:
+            logger.warning(
```

Comment:
This could be very noisy if you had a dataset with many long examples.  Consider capping the number of warnings, or lowering to an INFO

Reply:
changed to info, when we upgrade this model in future it'll be able to handle long context